### DISCUSSION TITLE
封闭在家，想学学xmake。

### DISCUSSION THREAD

**Discussion by dbyoung720** [xmake-io/xmake#2208] [General]
弱弱的问个问题：我想从下载、第三方依赖、编译、安装一次搞定，xmake 可以实现吗？

比如(Windows 平台下)：
  我想编译 libTIFF 库(还没有下载源码)，libTIFF 库又依赖 lzma、zlib 库，可以做到下载、第三方依赖库、编译、安装一次搞定吗？
  脚本该咋写？

补充：
  1、安装包中，放一些示例就好了。好像文档也没有？
  2、xmake-repo 好像都是示例吧，在 Windows 下如何运行？
  3、xrepo 可以运行。
    我试了一下，zlib 安装到 
    C:\Users\dbyoung\AppData\Local\.xmake\packages\z\zlib\v1.2.11\a593fa3f71e64cf790ec96898b34de8a 
    目录下面去了。这是直接下载编译好的库，不是编译？好像没看到编译过程。
   4、可以像 mxe 那样，可以编译某个平台下单个库，或者所有库？

**Comment by waruqi** [comment]
```lua
add_requires("libtiff")
target("test")
    add_files("src/*.cpp")
    add_packages("libtiff")
```

执行下 xmake 就知道了，readme, 文档，tboox.org 的文章，都有详细说明，仔细看下

**Comment by dbyoung720** [comment]
C:\Users\dbyoung\Desktop\tiff>xmake
checking for platform ... windows
checking for architecture ... x64
checking for Microsoft Visual Studio (x64) version ... 2022
note: install or modify (m) these packages (pass -y to skip confirm)?
in xmake-repo:
  -> libtiff v4.3.0 [vs_runtime:"MT"]
please input: y (y/n/m)
y
  => download https://gitlab.com/libtiff/libtiff/-/archive/v4.3.0/libtiff-v4.3.0.tar.gz .. ok
  => install libtiff v4.3.0 .. ok
[ 33%]: linking.release test.exe
error: LINK : warning LNK4001: 未指定对象文件；已使用库
LINK : fatal error LNK1561: 必须定义入口点

**Reply by waruqi** [reply]
你写 c++ 代码，连个 .cpp 文件都不加的么？。。不要照搬。。

**Comment by dbyoung720** [comment]
我需要写一个文件放到当前 src 目录下：
add_files("src/*.cpp")
是这个意思？
